federation/pkg/federation-controller/sync/controller.go

```diff
@@ -319,6 +319,13 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 		return redeliver
 	}
 
+	if _, ok := s.adapter.(federatedtypes.SchedulerAdapter); ok {
+		reconcileStatus := s.updateFedSpec(obj)
+		if reconcileStatus != statusOK {
+			return reconcileStatus
+		}
+	}
+
 	return syncToClusters(
```

Comment:
Yes, If you read the comments discussion in this PR, all this job can be done in one call to Schedule() (or whatever name might be apt for it). But I did keep an additional method for it, named FedUpdateStatus() currently. Its the adapters job to update the status of the Federation object.